llvm/lib/Support/Mustache.cpp

```diff
@@ -0,0 +1,489 @@
+      Lambda &L = Lambdas[Accessor[0]];
+      Value LambdaResult = L();
+      StringRef LambdaStr = printJson(LambdaResult);
+      Parser P = Parser(LambdaStr, Allocator);
+      ASTNode *LambdaNode = P.parse();
+      DenseMap<char, StringRef> EmptyEscapes;
+      return LambdaNode->render(Data, Allocator, Partials, Lambdas,
+                                SectionLambdas, EmptyEscapes);
+    }
+    return printJson(Context);
+  }
+  case Section: {
+    // Sections are not rendered if the context is falsey
+    bool IsLambda = SectionLambdas.find(Accessor[0]) != SectionLambdas.end();
+    if (isFalsey(Context) && !IsLambda)
+      return Result;
+    if (IsLambda) {
+      SectionLambda &Lambda = SectionLambdas[Accessor[0]];
+      Value Return = Lambda(RawBody);
+      if (isFalsey(Return))
+        return Result;
+      StringRef LambdaStr = printJson(Return);
+      Parser P = Parser(LambdaStr, Allocator);
+      ASTNode *LambdaNode = P.parse();
+      Result = LambdaNode->render(Data, Allocator, Partials, Lambdas,
+                                  SectionLambdas, Escapes);
+      return Result;
+    }
+    if (Context.getAsArray()) {
+      json::Array *Arr = Context.getAsArray();
+      for (Value &V : *Arr) {
+        for (ASTNode *Child : Children)
+          Result += Child->render(V, Allocator, Partials, Lambdas,
+                                  SectionLambdas, Escapes);
+      }
+      return Result;
+    }
+    for (ASTNode *Child : Children)
+      Result += Child->render(Context, Allocator, Partials, Lambdas,
+                              SectionLambdas, Escapes);
+    return Result;
+  }
+  case InvertSection: {
+    bool IsLambda = SectionLambdas.find(Accessor[0]) != SectionLambdas.end();
+    if (!isFalsey(Context) || IsLambda)
+      return Result;
+    for (ASTNode *Child : Children)
+      Result += Child->render(Context, Allocator, Partials, Lambdas,
+                              SectionLambdas, Escapes);
+    return Result;
+  }
+  }
+  llvm_unreachable("Invalid ASTNode type");
+}
+
+Value ASTNode::findContext() {
+  // The mustache spec allows for dot notation to access nested values
+  // a single dot refers to the current context
+  // We attempt to find the JSON context in the current node if it is not found
+  // we traverse the parent nodes to find the context until we reach the root
+  // node or the context is found
+  if (Accessor.empty())
+    return nullptr;
+  if (Accessor[0] == ".")
+    return LocalContext;
+  json::Object *CurrentContext = LocalContext.getAsObject();
+  StringRef CurrentAccessor = Accessor[0];
+  ASTNode *CurrentParent = Parent;
+
+  while (!CurrentContext || !CurrentContext->get(CurrentAccessor)) {
+    if (CurrentParent->T != Root) {
+      CurrentContext = CurrentParent->LocalContext.getAsObject();
+      CurrentParent = CurrentParent->Parent;
+      continue;
+    }
+    return nullptr;
+  }
+  Value Context = nullptr;
+  for (auto E : enumerate(Accessor)) {
```

Comment:
```suggestion
  for (auto [Idx, Acc]  : enumerate(Accessor)) {
```
It's probably easier to work with structured bindings.